sansa-rdf-spark/src/test/scala/net/sansa_stack/rdf/spark/model/df/DFTripleOpsTests.scala

```diff
@@ -0,0 +1,224 @@
+
+  test("difference of two RDF graph should result in size 0") {
+    val path = getClass.getResource("/loader/data.nt").getPath
+    val lang: Lang = Lang.NTRIPLES
+
+    val triples = spark.read.rdf(lang)(path)
+
+    val other = spark.read.rdf(lang)(path)
+
+    val graph = triples.difference(other)
+
+    val size = graph.count()
+
+    assert(size == 0)
+  }
+
+  test("intersection of two RDF graph should result in size 9") {
+    val path = getClass.getResource("/loader/data.nt").getPath
+    val lang: Lang = Lang.NTRIPLES
+
+    val triples = spark.read.rdf(lang)(path)
+
+    val other = spark.read.rdf(lang)(path)
+
+    val graph = triples.intersection(other)
+
+    val size = graph.count()
+
+    assert(size == 9)
+  }
+
+  test("add a statement to the RDF graph should result in size 10") {
+    val path = getClass.getResource("/loader/data.nt").getPath
+    val lang: Lang = Lang.NTRIPLES
+
+    val triple = Triple.create(
+      NodeFactory.createURI("http://dbpedia.org/resource/Guy_de_Maupassant"),
+      NodeFactory.createURI("http://xmlns.com/foaf/0.1/givenName"),
+      NodeFactory.createLiteral("Guy De"))
+
+    val triples = spark.read.rdf(lang)(path)
+
+    val graph = triples.add(triple)
+
+    val size = graph.count()
+
+    assert(size == 10)
+  }
+
+  test("add a list of statements to the RDF graph should result in size 12") {
+    val path = getClass.getResource("/loader/data.nt").getPath
+    val lang: Lang = Lang.NTRIPLES
+
+    val triple1 = Triple.create(
+      NodeFactory.createURI("http://dbpedia.org/resource/Guy_de_Maupassant"),
+      NodeFactory.createURI("http://xmlns.com/foaf/0.1/givenName"),
+      NodeFactory.createLiteral("Guy De"))
+
+    val triple2 = Triple.create(
+      NodeFactory.createURI("http://dbpedia.org/resource/Guy_de_Maupassant"),
+      NodeFactory.createURI("http://dbpedia.org/ontology/influenced"),
+      NodeFactory.createURI("http://dbpedia.org/resource/Tobias_Wolff"))
+
+    val triple3 = Triple.create(
+      NodeFactory.createURI("http://dbpedia.org/resource/Guy_de_Maupassant"),
+      NodeFactory.createURI("http://xmlns.com/foaf/0.1/givenName"),
+      NodeFactory.createURI("http://dbpedia.org/resource/Henry_James"))
+
+    val statements = Seq(triple1, triple2, triple3)
+
+    val triples = spark.read.rdf(lang)(path)
+
+    val graph = triples.addAll(statements)
+
+    val size = graph.count()
+
+    assert(size == 12)
+  }
+
+  test("remove a statement from the RDF graph should result in size 9") {
+    val path = getClass.getResource("/loader/data.nt").getPath
+    val lang: Lang = Lang.NTRIPLES
+
+    val triple = Triple.create(
+      NodeFactory.createURI("http://example.org/show/218"),
+      NodeFactory.createURI("http://www.w3.org/2000/01/rdf-schema#label"),
+      NodeFactory.createLiteral("That Seventies Show", "en"))
+
+    val triples = spark.read.rdf(lang)(path)
+    val graph = triples.remove(triple)
+
+    val size = graph.count()
+
+    assert(size == 9)
+  }
+  /*
```

Comment:
Either fix the following tests or remove them

Reply:
Sorry, I should put the comment few lines down. What I meant is that there are big comment block /**/, which should be removed (either the tests should work or just be removed)

Reply:
Guys, just a general comment. I don't think that having a `DataFrame` of Strings is efficient. I think the whole conversion from/to `Triple` is expensive.
Note, the benefits of having columns w.r.t. SPARQL/SQL is that fewer portion of the data might have to be touched.
What is the use-case/idea of having `Dataset` and `DataFrame` support? Why don'T why use a `DataFrame` of `Node` objects then? A possible converter/encoder would be:

```scala
implicit def toEncoded(t: Triple): JenaTripleEncoded = (t.getSubject, t.getPredicate, t.getObject)
implicit def fromEncoded(e: JenaTripleEncoded): Triple = Triple.create(e._1, e._2, e._3)

implicit def single[A](implicit c: ClassTag[A]): Encoder[A] = Encoders.kryo[A](c)

implicit def tuple3[A1, A2, A3](implicit e1: Encoder[A1], e2: Encoder[A2], e3: Encoder[A3]): Encoder[(A1, A2, A3)] =
      Encoders.tuple[A1, A2, A3](e1, e2, e3)
val triplesRDD: RDD[Triple] =  ...
val tripleNodesRDD = triplesRDD.map(t => (t.getSubject, t.getPredicate, t.getObject))
var tripleDS =
        session.createDataset(tripleNodesRDD)
      .toDF("s", "p", "o")
      .as[JenaTripleEncoded]

```
Without this encoding stuff, a `Triple` would just be a single binary column in a `Dataset`. You can check this with `tripleDS.printSchema()`:

```
root
 |-- s: binary (nullable = true)
 |-- p: binary (nullable = true)
 |-- o: binary (nullable = true)
```

I hope you get the idea.

Reply:
Yes, indeed. We have to find a solution to somehow encod the `Node` object into DataFrame, but on Dataset it is possible already (see here: https://github.com/SANSA-Stack/SANSA-RDF/blob/feature/df-tripleops/sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/rdd/TripleOps.scala#L42 ).  
And the use case is that we provide each of the Spark data structure (RDD/DataFrame/Dataset) with similar functions and users decide which one they prefer/are familiar to use. 

We could discuss this further and come with a more efficient way of conversation among different formats.

Reply:
On `Dataset` it's trivial. Nevertheless, and make sure that we're on the same level, `DataFrame` is nothing more than an alias for `Dataset[Row]`. Or to cite

> Dataset offers convenience of RDDs with the performance optimizations of DataFrames and the strong static type-safety of Scala.

Ideally, the triple would already be some case class such that we have columsn, but this is clearly not the case for Apache Jena `Triple` nor is it possible to hack this.

Nevertheless, I'd suggest to use `Node` instead of `String` as type for the columns in a `DataFrame`. This would at least avoid the parsing from and to `Triple` and allow for using the built-in functions regarding comparison RDF entities via `Node` methods.

Reply:
I think for now we will go with `DataFrame` of `String`, even having that on `DataFrame` of `Node` would have been better, but will have to investigate more how to define such `StructType` for Node and not just `binary` `DataType`.

Please, let me know what do you think about this. If you consider that for now, we go with `String`, I consider this PR ready to be merged (at least from my side).